src/components/MapView/Legends/index.tsx

```diff
@@ -27,14 +28,26 @@ function Legends({ classes, layers }: LegendsProps) {
   const isAnalysisLayerActive = useSelector(isAnalysisLayerActiveSelector);
 
   const legendItems = [
-    ...layers.map(({ title, legend, legendText }) => {
-      if (!legend || !legendText) {
+    ...layers.map(layer => {
+      if (!layer.legend || !layer.legendText) {
         // this layer doesn't have a legend (likely boundary), so lets ignore.
         return null;
       }
+
+      const wmsParams =
+        layer.type === 'wms' && layer.legend.length === 0
+          ? { baseUrl: layer.baseUrl, serverLayerName: layer.serverLayerName }
+          : undefined;
+
       return (
-        <LegendItem classes={classes} key={title} title={title} legend={legend}>
-          {legendText}
+        <LegendItem
+          classes={classes}
+          key={layer.title}
+          title={layer.title}
+          legend={layer.legend}
+          wmsParams={wmsParams}
```

Comment:
Thought - Maybe we could call this imageUrl instead and allow users to input an imageUrl instead of an array of color/text as legends? Think it could be useful?

Reply:
Included in 7dc68231018f6546fc2fd499d4ee6a68b0d3d36c